src/LogIndexOperationsTrait.php

```diff
@@ -0,0 +1,228 @@
+    }
+
+    private function logAddingDocumentToQueue(
+        LoggerInterface $logger,
+        IndexDocument $document,
+        string $indexName
+    ): void
+    {
+        $logger->info(
+            ($document ? sprintf('[documentId: "%s"] ', $document->getId()) : '') .
+            sprintf(
+                'Queuing document with types %s into index "%s"',
+                implode(', ', $document->getBody()['type'] ?? []),
+                $indexName
+            )
+        );
+    }
+
+    private function logMappings(LoggerInterface $logger, IndexDocument $document): void
+    {
+        if (!($this instanceof IndexerInterface)) {
+            return;
+        }
+
+        foreach (self::AVAILABLE_INDEXES as $documentType => $indexName) {
+            $logger->warning(
+                sprintf(
+                    'documentId: "%s" Index mappings: type="%s" index="%s"',
+                    $document->getId(),
+                    $documentType,
+                    $indexName)
+            );
+        }
+    }
+
+    private function debug(LoggerInterface $logger,  ?IndexDocument $document, string $message, ...$args): void
+    {
+        $logger->debug(
+            ($document ? sprintf('[documentId: "%s"] ', $document->getId()) : '') .
+            vsprintf($message, $args)
+        );
+    }
+
+    private function info(LoggerInterface $logger, ?IndexDocument $document, string $message, ...$args): void
+    {
+        $logger->info(
+            ($document ? sprintf('[documentId: "%s"] ', $document->getId()) : '') .
+            vsprintf($message, $args)
+        );
+    }
+
+    private function warn(LoggerInterface $logger, ?IndexDocument $document, string $message, ...$args): void
+    {
+        $logger->warning(
+            ($document ? sprintf('[documentId: "%s"] ', $document->getId()) : '') .
+            vsprintf($message, $args)
+        );
+    }
+
+    private function logErrorsFromResponse(LoggerInterface $logger, ?IndexDocument $document, $clientResponse): void
+    {
+        if (isset($clientResponse['errors']) && $clientResponse['errors']) {
+            $logger->warning(
+                ($document ? sprintf('[documentId: "%s"] ', $document->getId()) : '') .
+                sprintf(
+                    'Unexpected error response from client: %s',
+                    json_encode($clientResponse)
+                )
+            );
+        }
+    }
+
+    private function logUnclassifiedDocument(
+        LoggerInterface $logger,
+        ?IndexDocument $document,
+        string $method,
+        string $index,
+        $type = null,
+        $parentClasses = null
+    ): void
+    {
+        $this->logSkippedUpdate(
+            $logger,
+            'unclassified document',
+            $document,
+            $method,
+            $index,
+            $type,
+            $parentClasses
+        );
+    }
+
+    private function logSkippedUpdate(
+        LoggerInterface $logger,
+        string $why,
```

Comment:
Maybe it's better to call it `$reason` ?

Reply:
You're right, renamed in bfc406f606a0fb0f140a8a3ac666cd5b934948c7. Thanks!